File: src/main/java/SimpleObservable.java
import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class SimpleObservable {
    public static void main(String[] args) throws Exception {
        Observable<String> observable = Observable.create(emitter -> {
            System.out.println(
                Thread.currentThread().getName() + ":start sending"
            );

            String[] strings = {"o001", "o002", "o003", "o004", "o005"};

            for (String string : strings) {
                if (emitter.isDisposed()) {
                    return;
                }
                emitter.onNext(string);
            }

            emitter.onComplete();
        });

        observable.observeOn(Schedulers.computation()).subscribe(new Observer<String>() {
            @Override
            public void onSubscribe(Disposable d) {
                System.out.println(
                    Thread.currentThread().getName() + ": Ovservale start"
                );
            }

            @Override
            public void onNext(String s) {
                System.out.println(
                    Thread.currentThread().getName() + ": " + s
                );
            }

            @Override
            public void onError(Throwable e) {
                e.printStackTrace();
            }

            @Override
            public void onComplete() {
                System.out.println(
                    Thread.currentThread().getName() + ": onComplete"
                );
            }
        });

        Thread.sleep(500L);
    }
}

File: src/main/kotlin/SimpleObservable.kt
import io.reactivex.Observable
import io.reactivex.Observer
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import kotlin.system.exitProcess

fun main() {
    val observable = Observable.create<String> { emitter ->
        println(
            Thread.currentThread().name + ":start sending"
        )

        val strings = arrayOf("o001", "o002", "o003", "o004", "o005")

        for (string in strings) {
            if (emitter.isDisposed) {
                exitProcess(0)
            }
            emitter.onNext(string)
        }

        emitter.onComplete()
    }

    observable.observeOn(Schedulers.computation()).subscribe(object : Observer<String> {
        override fun onSubscribe(d: Disposable) {
            println(
                Thread.currentThread().name + ": Ovservale start"
            )
        }

        override fun onNext(s: String) {
            println(
                Thread.currentThread().name + ": " + s
            )
        }

        override fun onError(e: Throwable) {
            e.printStackTrace()
        }

        override fun onComplete() {
            println(
                Thread.currentThread().name + ": onComplete"
            )
        }
    })

    Thread.sleep(500L)
}
